graph2tac/loader/predict_server.py

```diff
@@ -156,13 +156,14 @@ def __init__(self,
         }
 
         self._node_i_to_name = model.get_label_to_name()
+        self._node_i_to_ident = model.get_label_to_ident()
         self._node_i_in_spine = model.get_label_in_spine()
         self._num_train_nodes = len(self._node_i_to_name)
 
         del self._def_node_to_i # not usable without a coq_context
-        self._def_name_to_i = {
+        self._def_ident_to_i = {
             name : i
-            for i,name in enumerate(self._node_i_to_name)
+            for i,name in enumerate(self._node_i_to_ident)
             if i >= self._base_node_label_num and self._node_i_in_spine[i]
```

Comment:
I would expect `self._node_i_in_spine[i]` to go away (otherwise there is not much improvement here). You'll still have to deal with some collisions though.